metrics/sources/summary/summary.go

```diff
@@ -233,6 +240,64 @@ func (this *summaryMetricsSource) decodeContainerStats(podLabels map[string]stri
 	return containerMetrics
 }
 
+// decodePVCStats computes PVC metrics using volume stats of each pod.
+// It selects the volume stats of the pod with the latest scrape time.
+// Metric collection time is the earliest pod start time among all pods.
+// PVC stats are only available when the PVC is in use by at least one pod.
+func (this *summaryMetricsSource) decodePVCStats(metrics map[string]*MetricSet, nodeLabels map[string]string, pod *stats.PodStats) {
+	glog.V(9).Infof("Decoding PVC stats for pod %s in namespace %s", pod.PodRef.Name, pod.PodRef.Namespace)
+
+	for _, vol := range pod.VolumeStats {
+		pvc := vol.PVCRef
+
+		if pvc == nil {
+			continue
+		}
+
+		pvcMetricName := PVCKey(pvc.Namespace, pvc.Name)
+		if _, ok := metrics[pvcMetricName]; !ok {
+			glog.V(9).Infof("Adding new metrics for PVC %s in namespace %s, using info from pod %s",
+				pvc.Name, pvc.Namespace, pod.PodRef.Name)
+
+			labels := map[string]string{
+				LabelMetricSetType.Key: MetricSetTypePVC,
+				LabelPVCName.Key:       pvc.Name,
+				LabelNamespaceName.Key: pvc.Namespace,
+				LabelPodRefs.Key:       pod.PodRef.Name,
+			}
+
+			pvcMetrics := &MetricSet{
+				Labels:              labels,
+				MetricValues:        map[string]MetricValue{},
+				LabeledMetrics:      []LabeledMetric{},
+				CollectionStartTime: pod.StartTime.Time,
+				ScrapeTime:          vol.Time.Time,
+			}
+
+			this.decodeFsStats(pvcMetrics, PVCResourcePrefix+pvc.Name, &vol.FsStats)
+			metrics[pvcMetricName] = pvcMetrics
+
+		} else {
+			labels := metrics[pvcMetricName].Labels
+			labels[LabelPodRefs.Key] = appendPodRef(labels[LabelPodRefs.Key], pod.PodRef.Name)
```

Comment:
nope, don't do this.  Leave it broken up per-pod.  People can aggregate in their sinks if they want it aggregated, but it's difficult-to-impossible to "unaggregate"